File: app/src/main/java/com/stupidtree/hichat/ui/relationevent/ResponseFriendTrigger.java
package com.stupidtree.hichat.ui.relationevent;

import androidx.annotation.NonNull;

import com.stupidtree.hichat.data.model.RelationEvent;
import com.stupidtree.hichat.ui.base.Trigger;

class ResponseFriendTrigger extends Trigger {
    String eventId;
    RelationEvent.ACTION action;


    public RelationEvent.ACTION getAction() {
        return action;
    }

    public String getEventId() {
        return eventId;
    }

    public static ResponseFriendTrigger getActioning(@NonNull String eventId, RelationEvent.ACTION action){
        ResponseFriendTrigger rft = new ResponseFriendTrigger();
        rft.action = action;
        rft.eventId = eventId;
        rft.setActioning();
        return rft;
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/ui/relationevent/ResponseFriendTrigger.kt
package com.stupidtree.cloudliter.ui.relationevent

import com.stupidtree.cloudliter.data.model.RelationEvent.ACTION
import com.stupidtree.cloudliter.ui.base.Trigger

internal class ResponseFriendTrigger : Trigger() {
    lateinit var eventId: String
    lateinit var action: ACTION

    companion object {
        fun getActioning(eventId: String, action: ACTION): ResponseFriendTrigger {
            val rft = ResponseFriendTrigger()
            rft.action = action
            rft.eventId = eventId
            rft.setActioning()
            return rft
        }
    }
}